### ISSUE TITLE
allow Python v3.11

### ISSUE THREAD

**Issue by dnelson86** [cbyrohl/scida#85]
Currently:

```bash
> pip install scida

  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-9dpfslea/numba_365e4a052da1401e902493b55ae552aa/setup.py", line 51, in <module>
          _guard_py_ver()
        File "/tmp/pip-install-9dpfslea/numba_365e4a052da1401e902493b55ae552aa/setup.py", line 48, in _guard_py_ver
          raise RuntimeError(msg.format(cur_py, min_py, max_py))
      RuntimeError: Cannot install on Python version 3.11.4; only versions >=3.7,<3.11 are supported.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Comment by dnelson86** [comment]
I see this comes from `numba`, and was resolved in 0.57, so really the `numba<0.57,>=0.56 (from scida)` requirement needs to be relaxed.

**Comment by cbyrohl** [comment]
correct, the numba>=0.56 requirement came from https://github.com/numba/numba/issues/8304 . Update to numba>=0.57 with next commit since problem with llvmlite seems to be resolved.
